<launch>
  <arg name="DEBUG" default="false" />
  <arg name="INPUT_POINT" default="/fisheye/input" /> <!--geometry_msgs/PointStamped-->
  <group ns="fisheye">
    <!-- relay fisheye image using compressed image -->
    <node pkg="image_transport" type="republish" name="chest_camera_remote" args="compressed">
      <remap from="in" to="/chest_camera/image_color" />
      <remap from="out" to="/chest_camera_remote/image_color" />
      <remap from="left/image_rect_color_decompressed" to="left/image_rect_color"/>
    </node>
    <node pkg="jsk_perception" type="fisheye_ray.py" name="fisheye_click_to_pose">
      <remap from="clicked_point" to="$(arg INPUT_POINT)"/>
      <remap from="camera_info" to="/chest_camera/camera_info"/>
      <remap from="image" to="/chest_camera_remote/image_color"/>
    </node>
    <node type="nodelet" pkg="nodelet" name="fisheye_manager" args="manager" />
    <!-- <node type="nodelet" pkg="nodelet" name="fisheye_to_panorama" -->
    <!--       args="load jsk_perception/FisheyeToPanorama fisheye_manager" output="screen"> -->
    <!--   <remap from="~input" to="/chest_camera/image_color"/> -->
    <!--   <param name="use_panorama" value="true"/> -->
    <!-- </node> -->
    <node type="nodelet" pkg="nodelet" name="fisheye_sphere"
          args="load jsk_pcl/FisheyeSpherePublisher fisheye_manager" output="screen">
      <remap from="~input" to="/chest_camera_remote/image_color"/>
    </node>
    <node type="image_view2" pkg="image_view2" name="fisheye_image_view" if="$(arg DEBUG)">
      <remap from="image" to="/chest_camera_remote/image_color"/>
      <remap from="/chest_camera_remote/image_color/screenpoint" to="$(arg INPUT_POINT)" />
    </node>
    <!-- <node type="image_view" pkg="image_view" name="undistorted_image_view"> -->
    <!--   <remap from="image" to="/fisheye_to_panorama/output"/> -->
    <!-- </node> -->
  </group>
</launch>
